pygeo/parameterization/DVGeoCST.py

```diff
@@ -0,0 +1,1093 @@
+        the upper and lower surfaces of the airfoil points, by default False
+    """
+
+    def __init__(self, datFile, idxChord=0, idxVertical=1, comm=MPI.COMM_WORLD, isComplex=False, debug=False):
+        super().__init__(datFile)
+        self.xIdx = idxChord
+        self.yIdx = idxVertical
+        self.comm = comm
+        self.isComplex = isComplex
+        if isComplex:
+            self.dtype = complex
+            self.dtypeMPI = MPI.DOUBLE_COMPLEX
+        else:
+            self.dtype = float
+            self.dtypeMPI = MPI.DOUBLE
+        self.debug = debug
+        if debug and not pltImport:
+            raise ImportError("matplotlib.pyplot could not be imported and is required for DVGeoCST debug mode")
+
+        # Store the DVs and flags to determine if the limited options have already been specified
+        self.DVs = {}
+        self.DVExists = {
+            "upper": False,
+            "lower": False,
+            "n1_upper": False,
+            "n2_upper": False,
+            "n1_lower": False,
+            "n2_lower": False,
+            "chord": False,
+        }
+
+        # Default DVs to be copied for each point set
+        self.defaultDV = {
+            "upper": 0.1 * np.ones(8, dtype=self.dtype),
+            "lower": -0.1 * np.ones(8, dtype=self.dtype),
+            "n1_upper": np.array([0.5], dtype=self.dtype),
+            "n2_upper": np.array([1.0], dtype=self.dtype),
+            "n1_lower": np.array([0.5], dtype=self.dtype),
+            "n2_lower": np.array([1.0], dtype=self.dtype),
+            "n1": np.array([0.5], dtype=self.dtype),
+            "n2": np.array([1.0], dtype=self.dtype),
+            "chord": np.array([1.0], dtype=self.dtype),
+        }
+
+        # ========== Process the input airfoil and set variables accordingly ==========
+        coords = readCoordFile(datFile)
+        self.foilCoords = np.zeros_like(coords, dtype=self.dtype)
+        self.foilCoords[:, self.xIdx] = coords[:, 0]
+        self.foilCoords[:, self.yIdx] = coords[:, 1]
+
+        # Set the leading and trailing edge x coordinates
+        self.xMin = np.min(self.foilCoords[:, self.xIdx])
+        self.xMax = np.max(self.foilCoords[:, self.xIdx])
+
+        # Check that the leading edge is at y = 0
+        idxLE = np.argmin(self.foilCoords[:, self.xIdx])
+        yLE = self.foilCoords[idxLE, self.yIdx]
+        if abs(yLE) > 1e-2:
+            raise ValueError(f"Leading edge y (or idxVertical) value must equal zero, not {yLE}")
+
+        # Determine the trailing edge thickness
+        idxTE = np.where(self.foilCoords[:, self.xIdx] == self.xMax)[0]
+        self.yUpperTE = np.max(self.foilCoords[idxTE, self.yIdx])
+        self.yLowerTE = np.min(self.foilCoords[idxTE, self.yIdx])
+        self.thicknessTE = self.yUpperTE - self.yLowerTE
+
+        # Compute splines for the upper and lower surfaces (used to split the foil in addPointSet)
```

Comment:
Good point, a subtle difference you found here that I didn't pick up on: prefoil splits with the LE as the point furthest from the TE, but  we also use the minimum x point for other operations (such as determining the chord). I don't think this is a problem though because there isn't overlap between where min x is used and where the prefoil split happens.